js_src/dismiss.ts

```diff
@@ -0,0 +1,40 @@
+export default class Dismiss {
+    private refersToElement;
+    private doesSoftDismiss;
+
+    public static softDismissDuration = 250; // ms
+
+    constructor(public element) {
+        this.refersToElement = document.querySelector(element.getAttribute("data-dismiss"));
+
+        this.doesSoftDismiss = element.hasAttribute("data-dismiss-soft");
+        
+        element.addEventListener("click", () => { this.dismiss() });
+    }
+
+    private dismiss() {
+        // Remove immediately, if no support for animate
+        if (!this.doesSoftDismiss || !this.refersToElement.animate) {
+            this.refersToElement.remove();   
+        } else {
+            this.refersToElement.animate([
+                { opacity: 1.0 },
+                { opacity: 0 }
+            ], {
+                duration: Dismiss.softDismissDuration
+            });
+            setTimeout(() => {
+                this.refersToElement.remove();
+            }, Dismiss.softDismissDuration)
+        }
+    }
+    
+    static find(query) {
+        const elements = document.querySelectorAll(query);
```

Comment:
Got to be careful with `querySelectorAll` here as it can return strange results sets in certain circumstances. I say this because of the public static method. Especially when the selector contains multiple selector attributes like `'.class1 .class2'` see https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll#User_notes

Reply:
This shouldn't be applicable here. It seems to be applicable to cases, where qSA() is applied to nodes AND has a hierarchy within the query. (At least the first one can't be, because it's on document)